test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Actions/UIActionEngineTests.cs

```diff
@@ -121,6 +123,57 @@ public async Task GetPreviewResultsAsync_WithMultipleActions_SortsPackageIdentit
             Assert.Equal(packageIdentityC.Id, addedResults[2].Id);
         }
 
+        [Fact]
+        public void AddUiActionEngineTelemetryProperties_AddsVulnerabilityInfo_Succeeds()
+        {
+            // Arrange
+            var telemetrySession = new Mock<ITelemetrySession>();
+            TelemetryEvent lastTelemetryEvent = null;
+            telemetrySession
+                .Setup(x => x.PostEvent(It.IsAny<TelemetryEvent>()))
+                .Callback<TelemetryEvent>(x => lastTelemetryEvent = x);
+
+            var operationId = Guid.NewGuid().ToString();
+
+            var actionTelemetryData = new VSActionsTelemetryEvent(
+                operationId,
+                projectIds: new[] { Guid.NewGuid().ToString() },
+                operationType: NuGetOperationType.Install,
+                source: OperationSource.PMC,
+                startTime: DateTimeOffset.Now.AddSeconds(-1),
+                status: NuGetOperationStatus.NoOp,
+                packageCount: 1,
+                endTime: DateTimeOffset.Now,
+                duration: .40);
+
+            UIActionEngine.AddUiActionEngineTelemetryProperties(
+                actionTelemetryEvent: actionTelemetryData,
+                continueAfterPreview: true,
+                acceptedLicense: true,
+                userAction: UserAction.CreateInstallAction("mypackageId", new NuGetVersion(1, 0, 0)),
+                selectedIndex: 0,
+                recommendedCount: 0,
+                recommendPackages: 0,
+                recommenderVersion: null,
+                topLevelVulnerablePackagesCount: 1,
+                topLevelVulnerableMaxSeverity: 1,
+                existingPackages: null,
+                addedPackages: null,
+                removedPackages: null,
+                updatedPackagesOld: null,
+                updatedPackagesNew: null,
+                targetFrameworks: null);
+
+            // Act
+            var service = new NuGetVSTelemetryService(telemetrySession.Object);
+            service.EmitTelemetryEvent(actionTelemetryData);
+
+            // Assert
+            Assert.NotNull(lastTelemetryEvent);
+            Assert.Equal(1, lastTelemetryEvent["topLevelVulnerableMaxSeverity"]);
+            Assert.Equal(1, lastTelemetryEvent["topLevelVulnerableMaxSeverity"]);
```

Comment:
these 2 lines looks similar. may be you want to assert for count of `topLevelVulnerablePackagesCount` also

```cs
Assert.Equal(1, lastTelemetryEvent["topLevelVulnerablePackagesCount"]);
```

Reply:
Yep. Fixed.